src/main/java/emu/grasscutter/server/packet/recv/HandlerHitTreeNotify.java

```diff
@@ -0,0 +1,69 @@
+package emu.grasscutter.server.packet.recv;
+
+import java.util.HashMap;
+
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.game.entity.EntityItem;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.game.world.Scene;
+import emu.grasscutter.net.packet.Opcodes;
+import emu.grasscutter.net.packet.PacketHandler;
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.VectorOuterClass;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.net.proto.HitCollisionOuterClass.HitCollision;
+import emu.grasscutter.utils.AutoRecycleHashMap;
+import emu.grasscutter.utils.Position;
+
+@Opcodes(PacketOpcodes.HitTreeNotify)
+public class HandlerHitTreeNotify extends PacketHandler {
+    private final static int RECORD_EXPIRED_SECONDS = 60*5;
+    private final static AutoRecycleHashMap<Integer,HashMap<Integer,HitTreeRecord>> hitRecord = new AutoRecycleHashMap<>(RECORD_EXPIRED_SECONDS);
+    @Override
+    public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+        HitCollision hit = HitCollision.parseFrom(payload);
+        VectorOuterClass.Vector hitPosition = hit.getHitPoint();
+        int woodType = hit.getHitColliderTypeValue();
+        if(woodType>=1 && woodType<=12) {// is a available wood type

```

Comment:
I don't see how this is supposed to work. `HitColliderType` does not tell you the type of wood you hit. It, as the name suggests, tells you the type of collider/hitbox.

Reply:
> I don't see how this is supposed to work. `HitColliderType` does not tell you the type of wood you hit. It, as the name suggests, tells you the type of collider/hitbox.

but I found there is regular pattern that ColliderTypeValue is 1 to 12, each is in 101301,101302,101303..etc,Do you have a better way to know which type I hit wood?

Reply:
I don't see how that would be possible. The return value of `getHitColliderTypeValue()` should only ever take values from -1 to 3, corresponding to the enum values defined in `emu.grasscutter.net.proto.HitColliderTypeOuterClass.HitColliderType`. And when I tested your code while doing my review, that is exactly the behavior I got. In fact, the value returned was always zero.

Reply:
> I don't see how that would be possible. The return value of `getHitColliderTypeValue()` should only ever take values from -1 to 3, corresponding to the enum values defined in `emu.grasscutter.net.proto.HitColliderTypeOuterClass.HitColliderType`. And when I tested your code while doing my review, that is exactly the behavior I got. In fact, the value returned was always zero.

Oh, I see, I used the wrong proto file, but it happened to resolve the type of wood. I'll re analyze the protocol!

> I don't see how that would be possible. The return value of `getHitColliderTypeValue()` should only ever take values from -1 to 3, corresponding to the enum values defined in `emu.grasscutter.net.proto.HitColliderTypeOuterClass.HitColliderType`. And when I tested your code while doing my review, that is exactly the behavior I got. In fact, the value returned was always zero.

Oh, I see, I used the wrong proto file, but it happened to resolve the type of wood. I'll re analyze the protocol!